docs/scripting.md

```diff
@@ -260,6 +260,26 @@ module.exports = (robot) ->
     res.send res.random leaveReplies
 ```
 
+## Custom Listeners
+
+While the above helpers cover most of the functionality the average user needs (hear, respond, enter, leave, topic), sometimes you would like to have very specialized matching logic for listeners. If so, you can use `listen` to specify a custom match function instead of a regular expression.
+
+The match function must return a truthy value if the listener callback should be executed. The truthy return value of the match function is then passed to the callback as response.match.
```

Comment:
I just realized that `message.user.name is "Steve" and Math.random() > 0.8` actually returns the random number (because it is the last thing in the expression), so `response.match` is already taken care of. Is just using that too clever for documentation?